src/camera.py

```diff
@@ -0,0 +1,266 @@
+                                       self._tilt_offset,
+                                       self.MIN_DEGREE,
+                                       self.MAX_TILT_DEGREE))
+
+
+    def calibrate_tilt(self, degree):
+        """
+        Calibrates the camera's tilting servo.
+
+        Writes a new tilt offset value to the config file and changes the current
+        tilt offset. The calibration value can be negative.
+
+        :param degree: The number of degrees.
+        :type degree: int
+        """
+        if self._camera_type == ABSOLUTE:
+            self._tilt_offset = int(np.clip(degree, self.MIN_TILT_CALI_DEGREE,
+                                                    self.MAX_TILT_CALI_DEGREE))
+        else:
+            self._tilt_offset = int(np.clip(degree + self._tilt_offset,
+                                                     self.MIN_TILT_CALI_DEGREE,
+                                                     self.MAX_TILT_CALI_DEGREE))
+
+        self._db.set('tilt_offset', self._tilt_offset)
+
+
+    def get_tilt_angle(self):
+        """
+        Gets the current tilt angle of the camera.
+
+        :return: The current tilt angle (in degrees).
+        :rtype: int
+        """
+        return self._current_tilt_angle
+
+
+def main():
+    """
+    Demonstrates the camera's panning and tilting functionality.
+
+    Creates a camera object and pans from 0 to 180, then from 180 to 0. The
+    camera then tilts from 0 to 150, then 150 to 0.
+    """
+    camera_delay = 0.05
+    picar.setup()
+
+    # Creating the object using absolute camera style.(default)
+    camera = Camera()
+    print "Testing absolute camera style"
+
+    # Tests absolute camera panning.
+    print "Panning left."
+    for i in range(0, 181, 2):
+        camera.pan(i)
+        time.sleep(camera_delay)
+
+    print "Panning right."
+    for i in range(180, -1, -2):
+        camera.pan(i)
+        time.sleep(camera_delay)
+
+    # Moving the camera to the center for tilting.
+    camera.pan(90)
+    time.sleep(camera_delay)
+
+    # Tests absolute camera tilting.
+    print "Tilting up."
+    for i in range(0, 151, 2):
+        camera.tilt(i)
+        time.sleep(camera_delay)
+
+    print "Tilting down."
+    for i in range(150, -1, -2):
+        camera.tilt(i)
+        time.sleep(camera_delay)
+
+    # Resetting camera for relative panning and tilting.
+    camera.pan(0)
+    time.sleep(camera_delay)
+
+    # Creating the object using relative camera style.
+    camera = Camera(camera_type=RELATIVE)
+    print "Testing relative camera style."
+
+    # Moving camera the camera to the far right before panning left.
+    camera.pan(-90)
+
+    # Tests relative camera panning..
+    print "Panning left."
+    for i in range(90):
+        camera.pan(2)
+        time.sleep(camera_delay)
+
+    print "Panning right."
+    for i in range(90):
+        camera.pan(-2)
+        time.sleep(camera_delay)
+
+    # Return camera to the center.
+    camera.pan(-90)
```

Comment:
Shouldn't this be just 90? If we panned from 0 to 180, then 180 to 0, -90 will keep it where it is.